### ISSUE TITLE
`verify clockSync3.qnt` fails with `Operator OPER_APP cannot be applied to arguments of types`

### ISSUE THREAD

**Issue by thpani** [informalsystems/apalache#2619]
A bug affecting the Quint -> Apalache transpilation.

ℹ️ I need to further investigate if this is a shortcoming on the Quint or Apalache side.
🗒️ For now, using this issue for note taking.

 In [`clockSync3.qnt`](https://github.com/informalsystems/quint/blob/f2994abce791cb2be5d8a67e9b5f51d77f12d7ba/examples/classic/distributed/ClockSync/clockSync3.qnt), we have

https://github.com/informalsystems/quint/blob/f2994abce791cb2be5d8a67e9b5f51d77f12d7ba/examples/classic/distributed/ClockSync/clockSync3.qnt#L87-L89

`msgs` has a type of `Set[{src : PROC, ts: int}]`, and `hc` is typed `PROC -> int`.

We also have

https://github.com/informalsystems/quint/blob/f2994abce791cb2be5d8a67e9b5f51d77f12d7ba/examples/classic/distributed/ClockSync/clockSync3.qnt#L34-L35

which gets typed `(a, b) => {src: a, ts: b}`.

So from `get/2` on `(hc, p)` we should infer `p: PROC` and the result of the application `get(hc, p) : int`.
The application result of `msgFrom(p, hc.get(p))` should be inferred as `{src: PROC, ts: int}`.

However, in the transpiled ApalacheIR, the type that reaches the typed builder is `{src: a, ts: b}`, causing it to fail with

```
Operator OPER_APP cannot be applied to arguments of types (((Set(a), a) => Set(a)), Set({ src: { id: Int }, ts: Int }), { src: a, ts: b })
```

**Comment by shonfeder** [comment]
I thought I had an issue open for this but I can’t find it. Iiuc, this is a symptom of the need to do fresh renaming of type variables when we combine type signatures from different places? Without that, we can end up with type variable collisions.

**Comment by thpani** [comment]
Jure fixed this in #2620.
We still need to consider the collision I described in https://github.com/informalsystems/apalache/issues/2619#issuecomment-1611332942, but that's a separate issue.
